### ISSUE TITLE
Pinning Nextcloud as PWA with Firefox opens Unsplash images in fullscreen

### ISSUE THREAD

**Issue by Valdiralita** [nextcloud/unsplash#34]
When pinning the Nextcould website as progressive web app onto your homescreen in android using firefox, the unsplash image opens as fullscreen image.

Does not happen with Chrome at all or regular Webpages in Firefox.

Android 8.1.0
Firefox 64.0.1
Nextcloud 14.0.4
Unsplash 1.1.3

Repro Steps:

Open Firefox and navigate to your Nextcloud page. 
Press the "house with plus" button next to the URL and add the website as "app" to your homescreen.
Go to your homescreen and open the new "app".

You should now see an Unsplash image everytime your URL/image changes.

Example Video:
(first webpage, then PWA)
https://www.youtube.com/watch?v=YQcC8nNGBAY

**Comment by jancborchardt** [comment]
@Valdiralita thanks a lot for the report, with video even! Could you test if this issue also happens with just theming? For that deactivate the Unsplash app, and in your Settings → Theming set a background image.

@newhinton @marius-wieschollek any idea here?

**Comment by marius-wieschollek** [comment]
I don't think we can do much here.

We use the url `https://source.unsplash.com/random/featured/?nature` in our stylesheet to get a random background. The url creates a 302 redirect to the actual image. That redirect seems to cause the issue. If i just use the target url in the css, the image is loaded correctly and no new tab is opened.

You can work around this issue by disabling the image in the navigation bar.

**Comment by Valdiralita** [comment]
You probably can expand the URL and follow the redirects, then put the result into css or set the background image directly.

**Comment by jancborchardt** [comment]
Then this is not trivial and directly related to:
- Request: Support other creative commons photo sites #5
- The Future of the Unsplash-App #33

TL;DR will likely only be fixed once we move to using actual APIs of services, like Wikimedia Commons.

**Comment by Valdiralita** [comment]
I can't reproduce this issue anymore. Maybe it was fixed with a new app/nextcloud version or with a firefox update.

**Comment by jancborchardt** [comment]
@Valdiralita thank you for the update, closing then! :)
